vision/medical_imaging/3d-unet-kits19/run.py

```diff
@@ -0,0 +1,176 @@
+# coding=utf-8
+# Copyright (c) 2021 NVIDIA CORPORATION. All rights reserved.
+# Copyright 2021 The MLPerf Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import sys
+sys.path.insert(0, os.getcwd())
+import argparse
+import subprocess
+
+from pathlib import Path
+
+import mlperf_loadgen as lg
+
+
+__doc__ = """
+Run 3D UNet performing KiTS19 Kidney Tumore Segmentation task.
+Dataset needs to be prepared through preprocessing (python preprocess.py --help).
+
+Run inference in performance mode:
+    python3 run.py --backend=$(BACKEND) --scenario=$(SCENARIO) --model=$(MODEL)
+
+Run inference in accuracy mode:
+    python3 run.py --backend=$(BACKEND) --scenario=$(SCENARIO) --model=$(MODEL) --accuracy
+
+$(BACKEND): tensorflow, pytorch, or onnxruntime
+$(SCENARIO): Offline, SingleStream, MultiStream, or Server (Note: MultiStream may be deprecated)
+$(MODEL) should point to correct model for the chosen backend
+
+If run for the accuracy, DICE scores will be summarized and printed at the end of the test, and 
+inference results will be stored as NIFTI files.
+
+Performance run can be more specific as:
+    python3  run.py --backend=$(BACKEND) --scenario=$(SCENARIO) --model=$(MODEL)
+                    --preprocessed_data_dir=$(PREPROCESSED_DATA_DIR)
+                    --postprocessed_data_dir=$(POSTPROCESSED_DATA_DIR)
+                    --mlperf_conf=$(MLPERF_CONF)
+                    --user_conf=$(USER_CONF)
+                    --performance_count=$(PERF_CNT)
+
+$(MLPERF_CONF) contains various configurations MLPerf-Inference needs and used to configure LoadGen
+$(USER_CONF) contains configurations such as target QPS for LoadGen and overrides part of $(MLPERF_CONF)
+$(PERF_CNT) sets number of query samples guaranteed to fit in memory
+
+More info for the above LoadGen related configs can be found at:
+https://github.com/mlcommons/inference/tree/master/loadgen
+"""
+
+
+def get_args():
+    """
+    Args used for running 3D UNet KITS19
+    """
+    parser = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument("--backend",
+                        choices=["pytorch", "onnxruntime", "tensorflow"],
+                        default="pytorch",
+                        help="Backend")
+    parser.add_argument("--scenario",
+                        choices=["SingleStream", "Offline",
```

Comment:
Would there be SingleStream/Server/MultiStream scenario for 3dunet?   I remember there is only offline.

Reply:
There are SingleStream and Offline, but no Server

Reply:
Thanks!

Reply:
Thanks - good point. I will remove unsupported scenarios from arg. :)